src/main/webapp/creatives/maps/SimidMap.js

```diff
@@ -0,0 +1,241 @@
+            this.playAd_();
+        }
+    }
+
+    /**
+     * Creates and displays a marker on the map representing a given place.
+     * @param {!Object} place A Place Result object.
+     * @private 
+     */
+    placeMapMarker_(place) {
+        const placeIcon = {
+            url: this.markerImage_,
+            scaledSize: new google.maps.Size(MARKER_SIZE, MARKER_SIZE)
+        };
+        const placeMarker = new google.maps.Marker({
+            map: this.map_,
+            position: place.geometry.location,
+            icon: placeIcon
+        });
+        ///Recalculate directions if a different active marker is selected.
+        placeMarker.addListener('click', () => {
+            this.activeLocation_ = place.geometry.location;
+            this.displayDirections_();
+        });
+    }
+
+    /**
+     * Displays the route between the starting loaction and destination
+     * based off of the selected travel mode.
+     * @private 
+     */
+    displayDirections_() {
+        this.directionsRenderer_.setMap(this.map_);
+        this.calculateRoute_();
+        this.calculateTravelTime_();
+        //If travel method changes, recalculate directions.
+        this.travelMethodElement_.addEventListener("change", () => {
+            this.calculateRoute_();
+            this.calculateTravelTime_();
+        });
+    }
+
+    /**
+     * Calculates the route between the user's current location and current
+     * active location based off of the selected travel mode.
+     * @private
+     */
+    calculateRoute_() {
+        const dirService = new google.maps.DirectionsService();
+        const selectedMode = this.travelMethodElement_.value;
+        dirService.route(
+            {
+                origin: this.currentLocation_,
+                destination: this.activeLocation_,
+                travelMode: [selectedMode]
+            },
+            (response, status) => {
+                if (status == "OK") {
+                    this.directionsRenderer_.setDirections(response);
+                } else {
+                    const directionsErrorMessage = {
+                        message: "ERROR: Failed to load directions: " + status,
+                    };
+                    //this.simidProtocol.sendMessage(CreativeMessage.LOG, directionsErrorMessage);
+                }
+            }
+        );
+    }
+
+    /**
+     * Calculates the time it takes to travel from the origin to the destination.
+     * @private
+     */
+    calculateTravelTime_() {
+        const travelMode = this.travelMethodElement_.value;
+        const matrixService = new google.maps.DistanceMatrixService();
+        matrixService.getDistanceMatrix(
+            {
+                origins: [this.currentLocation_],
+                destinations: [this.activeLocation_],
+                travelMode: [travelMode],
+                unitSystem: google.maps.UnitSystem.IMPERIAL
```

Comment:
If we don't hardcode a unitsystem in what happens?  Does it just default to the correct units gracefully?  Imperial is not an ideal unit outside USA.

Reply:
If we don't specify units the default is metric